sigopt_example.m

```diff
@@ -1,40 +1,51 @@
-% 
-%
-% define dimensions
-%
-% make connection object
-conn = py.sigopt_matlab.get_connection()
+import sigopt.Connection
 
-% dimension 1
-x_bounds = py.dict(pyargs('min', 0, 'max', 1))
-x_parameter = py.dict(pyargs('name', 'x', ...
-                             'type', 'double', ...
-                             'bounds', x_bounds))
-                        
-% dimension 2                
-y_bounds = py.dict(pyargs('min', 0, 'max', 1))
-y_parameter = py.dict(pyargs('name', 'y', ...
-                             'type', 'double', ...
-                             'bounds', y_bounds))
+conn = Connection('<YOUR API TOKEN HERE>')
 
-                    
-% define parameters and name variables
-parameters = py.list({x_parameter, y_parameter})
-name = 'Franke Optimization (MATLAB)'
+experiment = conn.experiments().create(struct( ...
```

Comment:
I think what Harvey is saying is that it is more Matlab-like to define an object with entries such that everything that can be is in a vector.  The idea of using dict-style objects is not the default.

I do think that, because we're allowing categoricals, we will need to keep things in the more struct-like setting.  We could also, instead, have a cell array that contains
```
prob.var_types = ['double', 'int', 'cat']
prob.bounds = {[0, 1], [1, 10], ['a', 'b', 'c']}
```
I think that Matlab folks are used to functions and tools having overloaded behavior, which I know is less desirable for what we do (I'm sure you've noticed me defaulting to the Matlab behavior at times since I've gotten here.

But, bottom line, I think the more Python style approach that you have written is fine.